src/generate/register.rs

```diff
@@ -166,16 +143,25 @@ pub fn render(
         }
     }
 
-    if access == Access::ReadOnly || access == Access::ReadWrite {
+    if read_access {
         mod_items.push(quote! {
             impl R {
                 #(#r_impl_items)*
             }
         });
     }
 
-    if access == Access::WriteOnly || access == Access::ReadWrite {
+    if write_access {
         mod_items.push(quote! {
+            /// Wiggle in the specified value into the given bits with mask and the offset and return the new value
+            #[inline]
+            pub const fn set_bits(bits: #rty, mask: #rty, offset: u8, value: #rty) -> #rty {
```

Comment:
@therealprof  I moved `set_bits` here. I think it is the best trade-off: one `const set_bits` per writable register as they can have different bit-size.